app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -281,6 +290,35 @@ class DefaultSessionControlController(
         )
     }
 
+    override fun handleRenameTopSiteClicked(topSite: TopSite) {
+        activity.let {
+            val customLayout =
+                    LayoutInflater.from(it).inflate(R.layout.top_sites_rename_dialog, null)
+            val topSiteLabelEditText: EditText =
+                    customLayout.findViewById(R.id.top_site_title)
+            topSiteLabelEditText.setText(topSite.title)
+
+            AlertDialog.Builder(it).apply {
+                setTitle(R.string.rename_top_site)
+                setView(customLayout)
+                setPositiveButton(R.string.top_sites_rename_dialog_ok) { _, _ ->
+                    val newTitle = topSiteLabelEditText.text.toString()
+                    if (newTitle.isNotBlank()) {
+                        viewLifecycleScope.launch(Dispatchers.IO) {
+                            with(activity.components.useCases.topSitesUseCase) {
+                                renameTopSites(topSite, newTitle)
+                            }
+                        }
+                    }
+                }
+                setNegativeButton(R.string.top_sites_rename_dialog_cancel, null)
```

Comment:
Looking at all our existing usages of `setNegativeButton` - I think we should be consistent, implement the dialog click listener and call `dialog.cancel()`